src/renderer/rendered_window.rs

```diff
@@ -152,69 +146,51 @@ impl RenderedWindow {
-            self.grid_current_position.x * font_dimensions.width as f32,
-            self.grid_current_position.y * font_dimensions.height as f32,
-        );
-
-        let image_size: (i32, i32) = (self.grid_size * font_dimensions).into();
-
-        Rect::from_point_and_size(current_pixel_position, image_size)
+    pub fn pixel_region(&self, grid_scale: GridScale) -> PixelRect<f32> {
+        GridRect::<f32>::from_origin_and_size(self.grid_current_position, self.grid_size.cast())
+            * grid_scale
     }
 
-    fn get_target_position(&self, outer_size: &Dimensions, padding_as_grid: &Rect) -> Point {
-        let destination = Point {
-            x: self.grid_destination.x + padding_as_grid.left,
-            y: self.grid_destination.y + padding_as_grid.top,
-        };
+    fn get_target_position(&self, grid_rect: &GridRect<f32>) -> GridPos<f32> {
+        let destination = self.grid_destination + grid_rect.min.to_vector();
 
         if self.anchor_info.is_none() {
             return destination;
         }
 
-        // Note the rect is always as far top/left as possible, which means that the right and
-        // bottom paddings might be bigger than requested. This is done in order to avoid the text
-        // moving around when the window is resized.
-        let valid_rect = Rect {
-            left: padding_as_grid.left,
-            right: padding_as_grid.left + outer_size.width as scalar,
-            top: padding_as_grid.top,
-            bottom: padding_as_grid.top + outer_size.height as scalar,
-        };
+        let mut grid_size: GridSize<f32> = self.grid_size.cast();
 
-        let mut grid_size = Point::new(self.grid_size.width as f32, self.grid_size.height as f32);
         if matches!(self.window_type, WindowType::Message { .. }) {
             // The message grid size is always the full window size, so use the relative position to
             // calculate the actual grid size
-            grid_size.y -= self.grid_destination.y;
+            grid_size.height -= self.grid_destination.y;
         }
-
+        // If a floating window is partially outside the grid, then move it in from the right, but
+        // ensure that the left edge is always visible.
         let x = destination
             .x
-            .min(valid_rect.right - grid_size.x)
-            .max(valid_rect.left);
+            .min(grid_rect.max.x - grid_size.width)
+            .max(grid_rect.min.x);
 
         // For messages the last line is most important, (it shows press enter), so let the position go negative
         // Otherwise ensure that the window start row is within the screen
-        let mut y = destination.y.min(valid_rect.bottom - grid_size.y);
-        if matches!(self.window_type, WindowType::Message { .. }) {
-            y = y.max(valid_rect.top)
+        let mut y = destination.y.min(grid_rect.max.y - grid_size.height);
+        if !matches!(self.window_type, WindowType::Message { .. }) {
```

Comment:
This was actually a bug, the condition was the wrong way. So the top of floating windows got pushed out. 

But there is still some other bug with this floating window repositioning code. The floating border gets clipped when resizing the window. This is not a regression here though, and I don't know the cause, so it will be fixed separately.